com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/test_props_url.lua

```diff
@@ -2,3 +2,7 @@ go.property("prop1", msg.url("url"))
 go.property("prop2", msg.url(""))
 go.property("prop3", msg.url())
 go.property("prop4", msg.url('url'))
+go.property("prop5", msg.url(''))
+go.property("prop6", msg.url('socket', '/path/to/object', 'fragment'))
+go.property("prop7", msg.url(hash('socket-hash'), hash('/path/to/object-hash'), hash('fragment-hash')))
+go.property("prop8", msg.url("", "", ""))
```

Comment:
Test cases from the editor and more (but see my comment here pls https://github.com/defold/defold/issues/9804#issuecomment-2504513482 )